### PR TITLE
[editor-common] Feat: Shift+Tab deletes tab char

### PR DESCRIPTION
**Pull request by maimonav**

### CODE REVIEW

File: packages/editor-common/web/src/Utils/draftUtils.js [hunk 1]
```diff
@@ -581,7 +581,30 @@ export function setForceSelection(editorState, selection) {
...
-  const selection = editorState.getSelection();
-  const newContentState = Modifier.replaceText(contentState, selection, string);
+  const selectionState = editorState.getSelection();
+  const newContentState = Modifier.replaceText(contentState, selectionState, string);
   return EditorState.push(editorState, newContentState, 'insert-string');
 }
+
+export function deleteTabCharacter(editorState) {
+  const contentState = editorState.getCurrentContent();
+  const selectionState = editorState.getSelection();
```
[the remark this answers is not in the document]
Comment (maimonav) [reply packages/editor-common/web/src/Utils/draftUtils.js:591]: fixed.

File: packages/editor-common/web/src/Utils/draftUtils.js [hunk 2]
```diff
@@ -581,7 +581,30 @@ export function setForceSelection(editorState, selection) {
...
+  //start offset of last character before cursor
+  const start = selectionState.getStartOffset() - 1;
```
[the remark this answers is not in the document]
Comment (maimonav) [reply packages/editor-common/web/src/Utils/draftUtils.js:592]: fixed.

File: packages/editor-common/web/src/Utils/draftUtils.js [hunk 3]
```diff
@@ -581,7 +581,39 @@ export function setForceSelection(editorState, selection) {
...
 export function insertString(editorState, string) {
   const contentState = editorState.getCurrentContent();
-  const selection = editorState.getSelection();
-  const newContentState = Modifier.replaceText(contentState, selection, string);
+  const selectionState = editorState.getSelection();
+  const newContentState = Modifier.replaceText(contentState, selectionState, string);
   return EditorState.push(editorState, newContentState, 'insert-string');
 }
+
+export function getLastCharacterFromSelection(editorState) {
```
Comment (shaulgo) [inline packages/editor-common/web/src/Utils/draftUtils.js:589]: rename to getCharacterBeforeSelection
getLastCharacterFromSelection means the last character in the selection.

File: packages/editor-common/web/src/Utils/draftUtils.js [hunk 4]
```diff
@@ -581,7 +581,39 @@ export function setForceSelection(editorState, selection) {
...
+  return character;
+}
+
+export function deleteTabCharacter(editorState) {
+  let newState;
+  const contentState = editorState.getCurrentContent();
+  const selectionState = editorState.getSelection();
+  if (selectionState.isCollapsed()) {
+    //start offset of last character before cursor
+    const start = selectionState.getStartOffset() - 1;
```
Comment (shaulgo) [inline packages/editor-common/web/src/Utils/draftUtils.js:609]: move to where it's needed

File: packages/editor-common/web/src/Utils/draftUtils.js [hunk 5]
```diff
@@ -581,7 +581,39 @@ export function setForceSelection(editorState, selection) {
...
+    const character = getLastCharacterFromSelection(editorState);
+
+    if (character === '\t') {
```
Comment (shaulgo) [inline packages/editor-common/web/src/Utils/draftUtils.js:612]: make only one if

File: packages/editor-common/web/src/Utils/draftUtils.js [hunk 6]
```diff
@@ -581,7 +581,39 @@ export function setForceSelection(editorState, selection) {
...
+      const newSelection = selectionState.set('anchorOffset', start);
+      const newContentState = Modifier.replaceText(contentState, newSelection, '');
+      newState = EditorState.push(editorState, newContentState, 'delete-string');
+    }
+  }
+  return newState;
```
Comment (shaulgo) [inline packages/editor-common/web/src/Utils/draftUtils.js:618]: you might return a null as editorState. Init the newState with editorState. 
Or don't return anything at all.

File: packages/editor-common/web/src/Utils/draftUtils.js (hunk 3 above)
Comment (maimonav) [reply packages/editor-common/web/src/Utils/draftUtils.js:589]: fixed.

File: packages/editor-common/web/src/Utils/draftUtils.js (hunk 5 above)
Comment (maimonav) [reply packages/editor-common/web/src/Utils/draftUtils.js:612]: fixed.

File: packages/editor-common/web/src/Utils/draftUtils.js (hunk 4 above)
Comment (maimonav) [reply packages/editor-common/web/src/Utils/draftUtils.js:609]: fixed.

File: packages/editor-common/web/src/Utils/draftUtils.js (hunk 6 above)
Comment (maimonav) [reply packages/editor-common/web/src/Utils/draftUtils.js:616]: That's good, I want to pass null, please look at HandleKeyCommand where I use this func.

File: packages/editor/web/src/RichContentEditor/handleKeyCommand.js [hunk 7]
```diff
@@ -15,20 +17,29 @@ const isList = blockType =>
...
+const handleTabCommand = (editorState, blockType, customHandlers, command) => {
+  let newState;
+  if (isList(blockType)) {
+    const direction = !event.shiftKey ? 1 : -1;
+    newState = indentSelectedBlocks(editorState, direction);
+  } else if (isTypeText(blockType)) {
+    if (!event.shiftKey) {
+      newState = insertString(editorState, CHARACTERS.TAB);
+    } else {
+      newState = deleteTabCharacter(editorState);
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/handleKeyCommand.js:29]: c = getCharchterBeforeSelection
if (c === tab) {
	deleteCharchterBeforeSelction
}

File: packages/editor-common/web/src/Utils/draftUtils.js (hunk 6 above)
Comment (shaulgo) [reply packages/editor-common/web/src/Utils/draftUtils.js:616]: I added a comment in HandleKeyCommand that solves this and hopefully makes everything nicer.

File: packages/editor/web/src/RichContentEditor/handleKeyCommand.js (hunk 7 above)
Comment (maimonav) [reply packages/editor/web/src/RichContentEditor/handleKeyCommand.js:29]: don't you should add the `isCollapse` as well?

File: packages/editor-common/web/src/Utils/draftUtils.js (hunk 6 above)
Comment (maimonav) [reply packages/editor-common/web/src/Utils/draftUtils.js:616]: fixed.

File: packages/editor-common/web/src/Utils/draftUtils.js [hunk 8]
```diff
@@ -581,7 +581,35 @@ export function setForceSelection(editorState, selection) {
...
+  if (start >= 0) {
+    const anchorKey = selectionState.getAnchorKey();
+    const contentState = editorState.getCurrentContent();
+    const currentContentBlock = contentState.getBlockForKey(anchorKey);
+    character = currentContentBlock.getText().slice(start, start + 1);
+  }
+  return character;
+}
+
+export function deleteTabCharacter(editorState) {
```
Comment (shaulgo) [inline packages/editor-common/web/src/Utils/draftUtils.js:603]: deleteCharacterCorsur

Comment (maimonav) [reply packages/editor-common/web/src/Utils/draftUtils.js]: fixed.

**Pull request merged**